pytensor/tensor/slinalg.py

```diff
@@ -1651,7 +1651,18 @@ class BlockDiagonal(BaseBlockDiagonal):
     def make_node(self, *matrices):
         matrices = self._validate_and_prepare_inputs(matrices, pt.as_tensor)
         dtype = _largest_common_dtype(matrices)
-        out_type = pytensor.tensor.matrix(dtype=dtype)
+
+        shapes_by_dim = tuple(zip(*(m.type.shape for m in matrices)))
```

Comment:
Unrelated to the changes but do you check that there is at least one matrix anywehere? Otherwise you may get a cryptic error at some point in make_node

Reply:
Yes, the `Op` is parameterized with `n_inputs`. 0 raises an error, and if you don't pass `n_inputs` matrices, it also raises an error.